src/main/java/org/elasticsearch/index/query/BaseQueryParser.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.search.Query;
+
+import java.io.IOException;
+
+public abstract class BaseQueryParser implements QueryParser {
```

Comment:
How about having a corresponding abstract class that the migrated queries can be moved to, which doesn't implement fromXContent and has this:

```
public final Query parse(QueryParseContext parseContext) throws IOException, QueryParsingException {
    return fromXContent(parseContext).toQuery(parseContext);
}
```

At the end of the migration, there will be no query extending this class, which can then be removed. No need for repeating the same `parse` impl all over. At some point later on we will also drop that other class, when `parse` can go away too